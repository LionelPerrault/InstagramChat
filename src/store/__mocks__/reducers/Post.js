import posts from './posts'

export default {
  feed: posts,
  posts,
  tagged: posts,
  shared: [
    {
      ...posts[0],
      share_id: 445,
      share_by_username: 'takkar',
      share_time: '1480114098767',
    },
  ],
  bookmarks: [posts[0]],
  photos: [
    {
      post_id: 43,
      user: 18,
      username: 'ragnar',
      firstname: 'Ragnar',
      surname: 'Lothbrok',
      post_time: '1480114098767',
      imgsrc: 'instagram_1516528062094.jpg',
      filter: 'filter-normal',
    },
  ],
  viewPost: {
    ...posts[0],
    comments: [
      {
        comment_id: 73,
        type: 'text',
        text: 'https://regexr.com/?37i6s fffffm',
        commentSrc: '',
        comment_by: 7,
        comment_by_username: 'ghalib',
        post_id: 89,
        comment_time: '1480114098767',
      },
      {
        comment_id: 69,
        type: 'image',
        text: '',
        commentSrc: 'instagram_comment_1518972851259.jpg',
        comment_by: 8,
        comment_by_username: 'coldplay',
        post_id: 89,
        comment_time: '1480114098767',
      },
      {
        comment_id: 104,
        type: 'sticker',
        text: '',
        commentSrc: 'instagram_comment_1527447929485.jpg',
        comment_by: 24,
        comment_by_username: 'takkar',
        post_id: 43,
        comment_time: '1480114098767',
      },
    ],
  },
  likes: [
    {
      like_id: 42,
      like_by: 18,
      username: 'ragnar',
      firstname: 'Ragnar',
      surname: 'Lothbrok',
      post_id: 43,
      like_time: '1480114098767',
      isFollowing: false,
    },
  ],
  tags: [
    {
      post_tag_id: 1,
      post_id: 89,
      user: 16,
      username: 'zayn',
      firstname: 'Zayn',
      surname: 'Malik',
      isFollowing: false,
    },
  ],
  isPostMine: false,
  usersToShare: [
    {
      follow_id: 302,
      follow_to: 16,
      username: 'zayn',
      firstname: 'Zayn',
      surname: 'Malik',
      didIShare: false,
    },
    {
      follow_id: 289,
      follow_to: 10,
      username: 'noddy',
      firstname: 'your',
      surname: 'noddy',
      didIShare: false,
    },
  ],
  sharers: [
    {
      share_id: 103,
      share_by: 7,
      share_by_username: 'ghalib',
      share_by_firstname: 'Mirza',
      share_by_surname: 'Ghalib',
      share_to: 28,
      share_to_username: 'selena',
      share_time: '1480114098767',
      post_id: 43,
      isFollowing: false,
    },
  ],
  postIt: {
    fileInput: '', // file input value
    fileChanged: false, // for checking file has changed
    targetFile: '', // file
    previewImg: '/images/location.jpg', // image which will be previewd
    desc: '', // textarea value
    filter: 'filter-normal',
    fetchingLocation: false,
    location: 'A-301, 90 Feet Road, Dharavi, Mumbai, Maharashtra 400017, India',
    addTag: false,
    tags: [
      {
        user: 7,
        username: 'ghalib',
      },
    ],
    showOverlay: false,
    type: 'user',
    group: 0,
  },
}
