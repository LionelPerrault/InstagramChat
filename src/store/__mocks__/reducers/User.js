export default {
  session: {
    id: 24,
    username: 'takkar',
  },
  user_details: {
    id: 24,
    username: 'takkar',
    firstname: 'iam_',
    surname: 'takkar',
    email: 'takkar@gmail.com',
    bio: '',
    account_type: 'public',
    email_verified: 'yes',
    joined: '1480114098767',
    twitter: '',
    facebook: '',
    github: 'https://github.com/yTakkar/',
    instagram: 'https://www.instagram.com/_faiyaz_shaikh',
    phone: '',
    website: '',
    isOnline: false,
    lastOnline: '',
  },
  tags: [
    {
      tag: 'traveler',
      user: 24,
    },
    {
      tag: 'nature-lover',
      user: 24,
    },
  ],
  mutualUsers: [
    {
      user: 11,
      username: 'nobita',
      firstname: 'nobita',
      surname: 'nobi',
      follow_id: 261,
      mutualUsersCount: 0,
    },
  ],
}
