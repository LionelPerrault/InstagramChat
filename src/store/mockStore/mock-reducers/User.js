export default {
  session: {
    id: 24,
    username: 'takkar',
  },
  user_details: {
    id: 7,
    username: 'ghalib',
    firstname: 'Mirza',
    surname: 'Ghalib',
    email: 'ghalib@gmail.com',
    bio: '',
    account_type: 'public',
    email_verified: 'yes',
    joined: '1514718748562',
    twitter: '',
    facebook: '',
    github: '',
    instagram: '',
    phone: '',
    website: '',
    isOnline: false,
    lastOnline: '',
  },
  tags: [
    {
      tag: 'traveler',
      user: 24,
    },
    {
      tag: 'nature-lover',
      user: 24
    }
  ]
}
