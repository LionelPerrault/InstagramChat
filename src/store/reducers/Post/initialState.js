import PostItInitialState from './postit-initial-state'

export default {
  posts: [],
  bookmarks: [],
  tagged: [],
  shared: [],
  photos: [],
  feed: [],
  viewPost: {},
  likes: [],
  tags: [],
  posted: false,
  isPostMine: false,
  usersToShare: [],
  sharers: [],
  postIt: PostItInitialState,
}
