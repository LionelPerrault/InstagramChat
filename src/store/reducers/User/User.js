/* eslint indent:0 */
/* eslint no-unreachable:0 */

import InitialState from './intialState'
import * as methods from './methods'

export default (state = InitialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USER_DETAILS':
      return {
        ...state,
        user_details: py.details,
        tags: py.tags,
      }
      break

    case 'ADD_TAG':
      return { ...state, tags: methods.addTag(state.tags, py) }
      break

    case 'DELETE_TAG':
      return { ...state, tags: methods.deleteTag(state.tags, py) }
      break

    case 'GET_MUTUAL_USERS':
      return { ...state, mutualUsers: py }
      break
  }

  return state
}
