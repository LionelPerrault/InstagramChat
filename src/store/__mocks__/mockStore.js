import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
