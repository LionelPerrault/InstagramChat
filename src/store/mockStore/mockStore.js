import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'
import Post from './mock-reducers/Post'
import Hashtag from './mock-reducers/Hashtag'
import Setting from './mock-reducers/Setting'
import Message from './mock-reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message
}

const middlewares = [
  promise(), thunk
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
