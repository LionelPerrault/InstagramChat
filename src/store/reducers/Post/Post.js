/* eslint indent:0 */
/* eslint no-unreachable:0 */

import initialState from './initialState'
import * as methods from './methods'

export default (state = initialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USER_POSTS':
      return { ...state, posts: py }
      break

    case 'GET_BOOKMARKED_POSTS':
      return { ...state, bookmarks: py }
      break

    case 'GET_TAGGED_POSTS':
      return { ...state, tagged: py }
      break

    case 'GET_SHARED_POSTS':
      return { ...state, shared: py }
      break

    case 'GET_PHOTOS':
      return { ...state, photos: py }
      break

    case 'GET_FEED':
      return { ...state, feed: py }
      break

    case 'GET_GROUP_POSTS':
      return { ...state, posts: py }
      break

    case 'GET_GROUP_PHOTOS':
      return { ...state, photos: py }
      break

    case 'GET_POST':
      return { ...state, viewPost: py }
      break

    case 'ADD_USER_POST':
      return {
        ...state,
        feed: methods.addPost(state.feed, py),
      }
      break

    case 'ADD_GROUP_POST':
      return {
        ...state,
        posts: methods.addPost(state.posts, py),
      }
      break

    case 'EDIT_POST':
      return {
        ...state,
        posts: methods.editPost(state.posts, py),
      }
      break

    case 'DELETE_POST':
      return {
        ...state,
        posts: methods.deletePost(state.posts, py),
        feed: methods.deletePost(state.feed, py),
      }
      break

    case 'GET_POST_LIKES':
      return {
        ...state,
        likes: py.likes,
        isPostMine: py.isPostMine,
      }
      break

    case 'REMOVE_LIKE':
      return {
        ...state,
        likes: methods.removeLike(state.likes, py),
      }
      break

    case 'GET_POST_TAGS':
      return {
        ...state,
        tags: py.tags,
        isPostMine: py.isPostMine,
      }
      break

    case 'UNTAG':
      return {
        ...state,
        tags: methods.untag(state.tags, py),
      }
      break

    case 'GET_USERS_TO_SHARE':
      return { ...state, usersToShare: py }
      break

    case 'GET_POST_SHARERS':
      return { ...state, sharers: py }
      break

    case 'UNBOOKMARK':
      return {
        ...state,
        bookmarks: methods.unbookmark(state.bookmarks, py),
      }
      break

    case 'REMOVE_SHARE':
      return {
        ...state,
        sharers: methods.removeShare(state.sharers, py),
      }
      break

    case 'COMMENT':
      return {
        ...state,
        viewPost: methods.comment(state.viewPost, py),
      }
      break

    case 'DELETE_COMMENT':
      return {
        ...state,
        viewPost: methods.deleteComment(state.viewPost, py),
      }
      break

    case 'EDIT_COMMENT':
      return {
        ...state,
        viewPost: methods.editComment(state.viewPost, py),
      }
      break

    // Post It
    case 'CHANGE_POSTIT_PROPS':
      return {
        ...state,
        postIt: methods.changePostIt(state.postIt, py),
      }

    case 'RESET_POSTIT':
      return {
        ...state,
        postIt: methods.resetPostItProperties(),
      }
  }

  return state
}
