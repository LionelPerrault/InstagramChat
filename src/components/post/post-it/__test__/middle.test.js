import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../store/mockStore/mockStore'
import PostItMiddle, { PurePostItMiddle } from '../middle'
import Post from '../../../../store/mockStore/mock-reducers/Post'
import { shallow } from 'enzyme'
import User from '../../../../store/mockStore/mock-reducers/User'

describe('PostItMiddle Component', () => {
  const comp = (
    <Provider store={mockStore}>
      <PostItMiddle/>
    </Provider>
  )

  it('should match snapshot when fileChanged = false', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should mock changeFile action', () => {
    const wrapper = shallow(
      <PurePostItMiddle
        session={User.session}
        postIt={Post.postIt}
        dispatch={jest.fn()}
      />
    )
    const file = new Blob(['foo'], {
      type: 'text/plain'
    })

    wrapper.find('FileInput').prop('fileChange')({
      preventDefault() {},
      target: {
        value: '',
        files: [file]
      }
    })
  })

  it('should match snapshot when fileChanged = false', () => {
    Post.postIt.fileChanged = true
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
