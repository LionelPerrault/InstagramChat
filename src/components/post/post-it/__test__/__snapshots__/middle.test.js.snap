// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItMiddle Component should match snapshot when fileChanged = false 1`] = `
<div
  className="i_p_main p_main"
  style={
    Object {
      "height": 296,
    }
  }
>
  <form
    className="post_img_form"
    encType="multipart/formdata"
    method="post"
  >
    <input
      accept="image/*"
      id="file_input"
      onChange={[Function]}
      type="file"
      value=""
    />
    <label
      className="pri_btn"
      htmlFor="file_input"
    >
      Choose an image
    </label>
  </form>
</div>
`;

exports[`PostItMiddle Component should match snapshot when fileChanged = false 2`] = `
<div
  className="i_p_main p_main"
  style={
    Object {
      "height": 296,
    }
  }
>
  <div>
    <div
      className="i_p_ta"
    >
      <textarea
        className="t_p_ta"
        onChange={[Function]}
        placeholder="What's new with you, @takkar?"
        spellCheck="false"
        value=""
      />
    </div>
    <div
      className="i_p_img"
    >
      <img
        className="filter-normal"
        src="/images/location.jpg"
      />
    </div>
  </div>
</div>
`;
